src/com/amazon/ion/IonSequence.java

```diff
@@ -410,6 +410,73 @@ public void add(int index, IonValue child)
      * was created.
      * </p>
      *
+     * The implementation of {@link List<IonValue>} returned by this method implements {@link Object#equals(Object)}
+     * and {@link Object#hashCode()}, with some caveats to be aware of illustrated below.
+     *
+     * Given:
+     *
+     * <code>
+     * int[] ints = new int[] {1, 2, 3, 4};
+     * IonList list = SYSTEM.newList(ints);
+     * IonSexp sexp = SYSTEM.newSexp(ints)
+     * IonSexp dgrm = SYSTEM.newDatagram(ints)
+     * List<IonValue> listSubList = list.subList(0, ints.size())
+     * List<IonValue> sexpSubList = sexp.subList(0, ints.size())
+     * List<IonValue> dgrmSubList = sexp.subList(0, ints.size())
+     * List<IonValue> arrayList = new ArrayList<IonValue>();
+     * for(int i : ints) { arrayList.add(SYSTEM.newInt(i)); }
+     * </code>
+     *
+     * {@link IonSequence#equals(Object)} always returns false when presented with a non {@link IonSequence}
+     * instance of {@link List<IonValue>}.  Hence, the following invocations of {@link Object#equals(Object)}
+     * return false even if the contained elements are equivalent.  This means that {@link Object#equals(Object)}
+     * is not fully symmetric.  The reason for the asymmetry is historical:  {@link IonSequence} has long violated the
+     * contract outlined by the {@link List} documentation.  Unfortunately, this cannot be changed because customers
+     * now rely on this behavior.
```

Comment:
I would reword the last sentence. Maybe 

"For the current major version of this library we  maintain backwards compatibility and support this behaviour as is."